src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1865,6 +1870,47 @@ ErrorIfQueryNotSupported(Query *queryTree)
 }
 
 
+/*
+ * VerifyDistinctClauseSupported returns true is provided query has supported
+ * distinct clause. Only target entries with Var, Const, Aggref types are supported
+ * at the moment.
+ * TODO : function support should be evaluated. immutable or stable functions can
+ * also be supported.
+ *
+ * TODO : Better error message is necessary at the caller.
+ */
+static bool
+VerifyDistinctClauseSupported(Query *query)
+{
+	List *targetList = query->targetList;
+	ListCell *targetCell = NULL;
+	List *distinctList = query->distinctClause;
+	ListCell *distinctCell = NULL;
+
+	Assert(query->distinctClause);
+
+	foreach(distinctCell, distinctList)
+	{
+		SortGroupClause *distinctGroupClause = (SortGroupClause *) lfirst(distinctCell);
+		Index sortGroupRef = distinctGroupClause->tleSortGroupRef;
+
+		foreach(targetCell, targetList)
+		{
+			TargetEntry *targetEntry = (TargetEntry *) lfirst(targetCell);
```

Comment:
Can we use `get_sortgroupclause_tle()` here?

Reply:
done